File: ActivitySuggestion/ManuallyReview/PhpClass/eventContainer.php
<?php 

require_once __DIR__ . '/../../utility.php';
require_once LIBPATH . '/simple_html_dom.php';

abstract class EventContainer{
	public $name;
	public $des; // description : string
	public $hostname; // string
	public $location; // string
	public $tel;
	public $GPS; // array length = 2, double
	public $poster; // string
	// public $originalContent; // string
	public $startDate; // string
	public $endDate; // string
	public $xmlStr; // string
	public $xmlObj; // simple html dom;
	public function __construct($xml){
		$this->xmlStr = $xml;
		$this->xmlObj = str_get_html($this->xmlStr);
	}
	public function ToJSON(){
		$phpArray = array(
			'name' => $this->name,
			'description' => $this->des,
			'hostname' => $this->hostname,  // string
			'location' => $this->location,  // string
			'tel' => $this->tel,  // string
			'GPS' => $this->GPS,  // array length = 2, double
			'poster' => $this->poster,  // string
			// 'originalContent' => $this->originalContent,  // string
			'startDate' => $this->startDate,  // string
			'endDate' => $this->endDate  // string
		);  // string
		return Utility::UnicodeJsonEncode($phpArray);
	}
	public abstract function Parse();
	/*
	public function constructWithXML($xml){
		$this->xmlStr = $xml;
		$this->xmlObj = str_get_html($this->xmlStr);
		return $this;
	}
	public function GetXmlDom(){
		if ($this->xmlObj){
			return $this->xmlObj;
		}else if ($this->xmlStr){
			$this->xmlObj = str_get_html($this->xmlStr);
			return $this->xmlObj;
		}
	}*/
}

?>

File: Crawler/src/EventContainer.java
import java.net.URL;
import java.sql.Date;


public class EventContainer {
	public String name;
	public String des; // description
	public String hostname;
	public String location;
	public double[] GPS; // length = 2
	public URL poster;
	public String originalContent;
	//public Date startDate;
	public String startDate;
	public String endDate;
	public EventContainer(){
		this.name = new String();
		this.des = new String();
		this.hostname = new String();
		this.location = new String();
		this.GPS = new double[2];
		// URL is not initially
		this.originalContent = new String();
		this.startDate = null;
	}
}
